apps/server/src/modules/learnroom/mapper/room-board-response.mapper.spec.ts

```diff
@@ -65,6 +65,33 @@ describe('room board response mapper', () => {
 			expect(result.elements[0] instanceof BoardElementResponse).toEqual(true);
 		});
 
+		it('should map tasks with status and its task card on board to response', () => {
+			const course = courseFactory.buildWithId();
+			const taskCard = taskCardFactory.buildWithId({ course });
+			const linkedTask = taskCard.task;
+			linkedTask.course = course;
+			linkedTask.taskCard = taskCard.id;
+			const status = {
+				graded: 0,
+				maxSubmissions: 0,
+				submitted: 0,
+				isDraft: false,
+				isFinished: false,
+				isSubstitutionTeacher: false,
+			};
+			const board = {
+				roomId: 'roomId',
+				displayColor: '#ACACAC',
+				title: 'boardTitle',
+				courseName: course.name,
+				elements: [{ type: RoomBoardElementTypes.TASK, content: { task: linkedTask, status } }],
+			};
+
+			const result = mapper.mapToResponse(board);
+
+			expect(result.elements[0] instanceof BoardElementResponse).toEqual(true);
```

Comment:
we miss an assertion specifically for taskCardId
```suggestion
			expect(result.elements[0] instanceof BoardElementResponse).toEqual(true);
			expect((result.elements[0].content as BoardTaskResponse).taskCardId).toEqual(taskCard.id);
```